internal/storage/bucket_handle.go

```diff
@@ -133,8 +133,20 @@ func (bh *bucketHandle) CreateObject(ctx context.Context, req *gcs.CreateObjectR
 	// We can't use defer to close the writer, because we need to close the
 	// writer successfully before calling Attrs() method of writer.
 	if err = wc.Close(); err != nil {
-		err = fmt.Errorf("error in closing writer: %v", err)
-		return
+		if err != nil {
+			switch ee := err.(type) {
+			case *googleapi.Error:
+				if ee.Code == http.StatusPreconditionFailed {
+					err = &gcs.PreconditionError{Err: ee}
+				}
+				if ee.Code == http.StatusNotFound {
```

Comment:
just curious when will createObject throw notfound error

Reply:
Sorry my bad, for create object it will not throw not found error.